node/tests/SharedTests.ts

```diff
@@ -2054,6 +2055,34 @@ export function runBaseTests<Context>(config: {
         config.timeout,
     );
 
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        `test blpop test_%p`,
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
```

Comment:
Use multiple keys here from the same hash slot
And add check for cross-slot scenario (see https://github.com/Bit-Quill/glide-for-redis/pull/226 and https://github.com/Bit-Quill/glide-for-redis/pull/196/)